sync-core/src/test/java/com/cloudant/sync/query/QuerySqlTranslatorTest.java

```diff
@@ -889,12 +919,31 @@ public void usesCorrectSQLOperatorWhenUsingNOTLTE() {
         SqlParts where = QuerySqlTranslator.whereSqlForAndClause(Arrays.<Object>asList(name),
                                                                  indexName);
         String expected = String.format("_id NOT IN (SELECT _id FROM %s WHERE \"name\" <= ?)",
-                                        IndexManager.tableNameForIndex(indexName));
+                                        indexTable);
         assertThat(where.sqlWithPlaceHolders, is(expected));
         assertThat(where.placeHolderValues, is(arrayContaining("mike")));
     }
 
     @Test
+    public void constructsValidWhereClauseForNOTIN() {
+        Map<String, Object> op = new HashMap<String, Object>();
+        op.put("$in", Arrays.<Object>asList("mike", "fred"));
+        Map<String, Object> notOp = new HashMap<String, Object>();
+        notOp.put("$not", op);
+        Map<String, Object> name = new HashMap<String, Object>();
+        name.put("name", notOp);
+
+        SqlParts where = QuerySqlTranslator.whereSqlForAndClause(Arrays.<Object>asList(name),
+                                                                 indexName);
+        String expectedSubWhere = "WHERE \"name\" IN ( ?, ? ))";
+        String expected = String.format("_id NOT IN (SELECT _id FROM %s %s",
+                                        indexTable,
+                                        expectedSubWhere);
```

Comment:
I'd prefer to see the whole sub-select statement as a separate variable. At the very least, have the balancing bracket for the `SELECT` in the second string.